drivers/pinctrl/intel/pinctrl-intel.c

```diff
@@ -1503,6 +1541,12 @@ int intel_pinctrl_suspend(struct device *dev)
 		base = community->regs + community->ie_offset;
 		for (gpp = 0; gpp < community->ngpps; gpp++)
 			communities[i].intmask[gpp] = readl(base + gpp * 4);
+
+		if (dmi_first_match(dmi_retain_hostown_table)) {
```

Comment:
Can we store a flag indicating that this device needs the quirk in the previous call to `dmi_first_match` and check that flag here instead of calling `dmi_first_match` again? Same for the following calls to `dmi_first_match`.

Reply:
@jprvita  Thanks. I've forcely updated the original commit to use a flag named `quirks` instead of `dmi_first_match`. Please leave comments if there's still something to improve. Thanks.